src/Microsoft.AspNetCore.Routing/Matchers/CandidateSet.cs

```diff
@@ -2,56 +2,149 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Runtime.CompilerServices;
 
 namespace Microsoft.AspNetCore.Routing.Matchers
 {
-    internal class CandidateSet
+    public sealed class CandidateSet
```

Comment:
Suggestions welcome on making this spicy boy more efficient.

Reply:
`boy` --> `boi`

Reply:
noice